### PR TITLE
fix: Sidepanel Modal mode for small screens

### LINKED ISSUE #20294: Bug: Feature preview opens on resize
## Bug description

If you visit the feature previews menu and then close it, it will automatically reopen when you resize the window. 

https://github.com/PostHog/posthog/assets/84011561/2cacfdef-e335-49b8-8f2e-0c0775064126

## How to reproduce

1. Open feature previews
2. Close it
3. Resize window

## Environment

- [ ] PostHog Cloud US, project ID: [please provide from https://app.posthog.com/settings/project-details#variables]
- [ ] PostHog Cloud EU, project ID: [please provide from https://eu.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see ["Sunsetting Kubernetes support"](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Fixes https://github.com/PostHog/posthog/issues/20294

We don't show the side panel for small screens, but we do sometimes still rely on it.

## Changes

* If we are in small layout, don't show the side panel but render open panels in a modal instead
* It's not perfect but its **#GoodEnough™️** 

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/__snapshots__/layout-feature-previews-modal--basic--dark.png
Comment (Twixes) [inline frontend/__snapshots__/layout-feature-previews-modal--basic--dark.png:1]: ❗

**Review by Twixes — COMMENTED** [review]
Do we still need [this](https://github.com/PostHog/posthog/blob/df461f14866b05751eeedd11d89cb1f7e2393507/frontend/src/lib/components/Support/supportLogic.ts#L214-L219) bit after the changes here?

**Comment by benjackwhite** [comment]
> Do we still need [this](https://github.com/PostHog/posthog/blob/df461f14866b05751eeedd11d89cb1f7e2393507/frontend/src/lib/components/Support/supportLogic.ts#L214-L219) bit after the changes here?

Yep. It's used in scene where we don't have any navigation at all like signup

File: frontend/__snapshots__/layout-feature-previews-modal--basic--dark.png
Comment (benjackwhite) [reply frontend/__snapshots__/layout-feature-previews-modal--basic--dark.png:1]: Not sure why that showed up given that the whole story is removed

**Pull request merged**